### ISSUE TITLE
LossEvaluator ignores the autocast context passed by Trainer

### ISSUE THREAD

**Issue by WhymustIhaveaname** [suning-git/nanoinfra#4]
## Problem

`Trainer._evaluate()` passes its BF16 autocast context to every evaluator, but the generic `LossEvaluator.evaluate()` names the argument `_autocast_ctx` and never enters it. Its logits and fused evaluation services therefore run outside the same autocast regime used for training.

This matters for custom `System` implementations containing FP32 trainable modules. In downstream use, evaluation had to wrap the evaluator-facing trunk manually in `torch.autocast` to make evaluation match training-time computation. Native systems whose parameters are already stored as BF16 can hide the problem.

## Minimal reproduction

```python
class TrackingContext:
    def __init__(self):
        self.entries = 0
    def __enter__(self):
        self.entries += 1
    def __exit__(self, *args):
        pass

ctx = TrackingContext()
evaluator = LossEvaluator(loader, eval_steps=1, mode="logits", total_metric="loss")
evaluator.evaluate(system, ctx)
assert ctx.entries == 1  # currently 0
```

The current path is:

```python
def evaluate(self, model, _autocast_ctx):
    ...
    raw = eval_fn(model, self.dataloader, self.eval_steps, ...)
```

## Expected behavior

The generic evaluator should enter the supplied context around model evaluation, or the interface should stop accepting the context and clearly require callers to wrap it. Since `Trainer` already supplies the context, entering it in `LossEvaluator` is the least surprising contract. A regression test should verify that the context is entered for both `logits` and `fused` modes.

`TextEvaluator` currently wraps its nested `LossEvaluator` externally; nested autocast contexts are harmless, though that wrapper could be simplified after the generic contract is fixed.

**Comment by WhymustIhaveaname** [comment]
Resolved in commit f23554c.

`LossEvaluator.evaluate()` now enters the autocast context the Trainer hands it, for both the logits and the fused path. `None` is accepted. The contract is stated in the `Evaluator` docstring: the Trainer passes the context, and the evaluator enters it around whatever should run under the training precision regime.

One detail worth noting from the fix: the context must be entered exactly once. `torch.amp.autocast` keeps its restore state on the instance, so nesting the same object leaves autocast switched on after the outer exit. `TextEvaluator` therefore dropped its own `with autocast_ctx` wrapper.

`core/tests/unit/test_evaluator_autocast.py` implements the counting-context reproduction from this issue and asserts exactly one entry per `evaluate()` in both modes.

Closing.
